### PR TITLE
cmd/libsnap-confine-private, cmd/snap-confine: fix logging of failed [u]mount when run with SNAP_CONFINE_DEBUG=1

### PR DESCRIPTION
**Pull request by bboozzoo**
The series fixes a failed [u]mount command not being logged when snap-confine is run with debug logging enabled.

The logic in `libsnap-confine-private.c:sc_do_[u]mount` would overwrite `[u]mount_cmd` with pointer to `(disabled) use debug build to see details` message, thus the later NULL check tests would fail. I've updated `sc_do_[u]mount` tests to cover this behavior.

Also, the `SNAP_CONFINE_DEBUG_BUILD` CFLAG does not propagate from `snap-confine-debug` to `libsnap-confine-private.a`. There are 2 options to fix this, one is to add a 'debug' variant of the library, or pull in the library source code directly into `snap-confine-debug`. I've opted for the second variant.

### CODE REVIEW

**Comment by mvo5** [comment]
This fails on Debian with: 
```
mv -f libsnap-confine-private/.deps/snap_confine_snap_confine_debug-utils.Tpo libsnap-confine-private/.deps/snap_confine_snap_confine_debug-utils.Po
make[3]: *** No rule to make target '-ludev', needed by 'snap-confine/snap-confine-debug'.  Stop.
```

File: cmd/libsnap-confine-private/mount-opt-test.c [hunk 1]
```diff
@@ -224,8 +227,14 @@ static void test_sc_do_mount(void)
...
 	g_test_trap_assert_failed();
-	g_test_trap_assert_stderr
-	    ("cannot perform operation: mount -t ext4 -o ro /foo /bar: Permission denied\n");
+	if (GPOINTER_TO_INT(snap_debug) == 0) {
+		g_test_trap_assert_stderr
+		    ("cannot perform operation: mount -t ext4 -o ro /foo /bar: Permission denied\n");
+	} else {
+		g_test_trap_assert_stderr
+		    ("DEBUG: performing operation: (disabled) use debug build to see details\n"
+		     "cannot perform operation: mount -t ext4 -o ro /foo /bar: Permission denied\n");
```
Comment (mvo5) [inline cmd/libsnap-confine-private/mount-opt-test.c]: Pardon my ignorance - but why do I see the real operation here still? Isn't the point of the non-debug build to hide this? Maybe a comment here that explains what is going on?

File: cmd/libsnap-confine-private/mount-opt-test.c [hunk 2]
```diff
@@ -247,15 +259,27 @@ static void test_sc_do_umount(void)
...
 	g_test_trap_assert_failed();
-	g_test_trap_assert_stderr
-	    ("cannot perform operation: umount --lazy /foo: Permission denied\n");
+	if (GPOINTER_TO_INT(snap_debug) == 0) {
+		g_test_trap_assert_stderr
+		    ("cannot perform operation: umount --lazy /foo: Permission denied\n");
+	} else {
+		g_test_trap_assert_stderr
+		    ("DEBUG: performing operation: (disabled) use debug build to see details\n"
+		     "cannot perform operation: umount --lazy /foo: Permission denied\n");
```
Comment (mvo5) [inline cmd/libsnap-confine-private/mount-opt-test.c]: Same question as above.

File: cmd/libsnap-confine-private/mount-opt-test.c (hunk 1 above)
Comment (bboozzoo) [reply cmd/libsnap-confine-private/mount-opt-test.c]: We expect it to be visible regardless of debug build and SNAP_CONFINE_DEBUG setting as it's useful for figuring out what failed exactly. Previously, in non debug builds, you'd only see the error to contain `mount ..` if you ran without SNAP_CONFINE_DEBUG. When running with SNAP_CONFINE_DEBUG, you'd get `(disabled) use debug build to see details` instead.

File: cmd/libsnap-confine-private/mount-opt.c [hunk 3]
```diff
@@ -251,6 +251,9 @@ const char *sc_umount_cmd(char *buf, size_t buf_size, const char *target,
 	return buf;
 }
 
+static const char use_debug_build[] =
```
Comment (zyga) [inline cmd/libsnap-confine-private/mount-opt.c]: Please use `static const char use_debug_build * = "..."`

File: cmd/libsnap-confine-private/mount-opt-test.c (hunk 1 above)
Comment (zyga) [reply cmd/libsnap-confine-private/mount-opt-test.c]: This was meant as a security precaution against running diagnostic code in production builds. If that code contains any errors it could be used to attack a setuid-root program.

File: cmd/libsnap-confine-private/mount-opt-test.c [hunk 4]
```diff
@@ -211,10 +211,13 @@ static bool broken_mount(struct sc_fault_state *state, void *ptr)
 	return true;
 }
 
-static void test_sc_do_mount(void)
```
Comment (zyga) [inline cmd/libsnap-confine-private/mount-opt-test.c]: Can you please spin this off into a separate PR and discuss with @jdstrand when he's back in two weeks?

File: cmd/Makefile.am [hunk 5]
```diff
@@ -102,6 +102,10 @@ libsnap_confine_private_a_SOURCES = \
 	libsnap-confine-private/utils.h
 libsnap_confine_private_a_CFLAGS = $(CHECK_CFLAGS)
```
Comment (zyga) [inline cmd/Makefile.am]: This is ok, thank you for doing that :)

File: cmd/libsnap-confine-private/mount-opt-test.c (hunk 1 above)
Comment (mvo5) [reply cmd/libsnap-confine-private/mount-opt-test.c]: I'm probably a bit slow and/or missing something - but if we show the real operation, why do we show the "diabled use debug build" message at all?

**Comment by bboozzoo** [comment]
Replaced by #4482 and #4483
